### PR TITLE
Add `RouteArgument` attribute for Yii Hydrator

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Review by xepozz — CHANGES_REQUESTED** [review]
`Route` is a wrong name for this kind of attribute.

`Route` is already reserved for routing: the class and coming attribute as well. 
I suggest you think about `RouteAttribute` or something similar. Do not confuse developers with one more `Route` entity.

**Comment by xepozz** [comment]
Why is it `argument`, isn't `parameter` better?

**Comment by vjik** [comment]
> Why is it `argument`, isn't `parameter` better?

In router, this is called "arguments": https://github.com/yiisoft/router/blob/0892cbebfdfbe73ae9d2a64a133c96daf1552c7a/src/CurrentRoute.php#L112

File: composer.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
         "vimeo/psalm": "^4.30|^5.20",
         "yiisoft/di": "^1.0",
         "yiisoft/dummy-provider": "^1.0.0",
+        "yiisoft/hydrator": "dev-master",
```
Comment (samdark) [inline composer.json:42]:
```suggestion
        "yiisoft/hydrator": "^1.0",
```

**Review by samdark — APPROVED** [review]

**Pull request merged**